### PR TITLE
In datm, set orbital parameters in run loop

### PR DESCRIPTION
**Pull request by billsacks**
This is needed for datm to restart properly when using variable-year
orbital parameters.

Test suite:
- scripts_regression_tests on cheyenne
- These tests now pass in CTSM tag clm5.0.dev010 if I put in a hack in
  CTSM to fix a problem there:
  - ERS_D_P36x1.f10_f10_musgs.IHistClm50Bgc.cheyenne_intel.clm-decStart
  - ERP_D_P48x1.f10_f10_musgs.IHistClm50Bgc.hobart_nag.clm-decStart
- These tests are bit-for-bit with out-of-the-box clm5.0.dev010
  (confirming that these changes haven't completely broken datm; note
  that this includes a selection of 1850, 2000 and HIST tests, but
  deliberately does not include HIST tests that cross the year boundary):
  - ERP_D_P36x2_Ld3.f10_f10_musgs.I1850Clm50BgcCrop.cheyenne_intel.clm-default1850
  - ERP_D_Ld3_P36x2.f10_f10_musgs.I2000Clm50BgcCruGs.cheyenne_intel.clm-default
  - ERP_D_Ld5.f10_f10_musgs.I2000Clm50Sp.cheyenne_intel.clm-decStart
  - ERS_D_Ld10.f10_f10_musgs.IHistClm50SpG.cheyenne_intel.clm-glcMEC_decrease

Test baseline: clm5.0.dev010
Test namelist changes: none
Test status: potentially climate-changing for I compsets that use
variable_year orbital parameters (currently HIST compsets), and any
other compsets using datm with solar zenith angle-based interpolation (I
*think* that's the criteria determining if these changes change
answers): Previously, datm looked at orbital parameters at the start of
the run, and used those orbital parameters for its zenith angle-based
interpolations throughout the entire run (though re-checking this on
each restart). Now, datm re-checks the orbital parameters every time
step, so the zenith angle-based interpolations are kept in sync with the
current orbital parameters.

Fixes ESMCI/cime#2598

User interface changes?: none

Update gh-pages html (Y/N)?: N

Code review:

### CODE REVIEW

**Comment by billsacks** [comment]
@rljacob - I'm adding you as a reviewer in case you want to look at this, because I think it might affect e3sm, too.

@mvertens - In addition to reviewing the diffs, can you also confirm my statement in the top comment about when this will change answers?

@ekluzek also bringing you into the loop

**Comment by rljacob** [comment]
Will it change answers for **any** case using datm?

File: src/components/data_comps/datm/datm_comp_mod.F90 [hunk 1]
```diff
@@ -312,9 +312,6 @@ subroutine datm_comp_init(Eclock, x2a, a2x, &
        k = mct_aVect_indexRA(SDATM%grid%data,'frac')
        SDATM%grid%data%rAttr(k,:) = 1.0_R8
 
-       !--- set data needed for cosz t-interp method ---
-       call shr_strdata_setOrbs(SDATM,orbEccen,orbMvelpp,orbLambm0,orbObliqr,idt)
```
Comment (billsacks) [inline src/components/data_comps/datm/datm_comp_mod.F90]: Note that I removed the setOrbs call from datm_comp_init. There is now a setOrbs call in datm_comp_run, and datm_comp_run is called from near the end of datm_comp_init, so this seemed like a safe thing to do. But I wasn't 100% confident in this being safe, which was part of my motivation for running a handful of tests to confirm I hadn't changed answers.

If others feel this removal is potentially problematic, I can put it back in.

**Comment by billsacks** [comment]
> Will it change answers for any case using datm?

For many cases, it will not change answers. First, it should only change answers for cases with variable_year orbital parameters. Second, my understanding (though I'd like confirmation from @mvertens ) is that only select datm modes have logic that relies on the orbital parameters (primarily the modes used for forcing a land model, I think - not the modes for forcing an ocean model, at least in CESM).

@rljacob and @mvertens do you feel that we should do more testing to more definitively characterize when this changes answers?

**Comment by billsacks** [comment]
@rljacob - a bit more about when/why it changes answers (I'll also add this in the big comment at the top): Previously, datm looked at orbital parameters at the start of the run, and used those orbital parameters for its zenith angle-based interpolations throughout the entire run (though re-checking this on each restart). Now, datm re-checks the orbital parameters every time step, so the zenith angle-based interpolations are kept in sync with the current orbital parameters.

**Comment by billsacks** [comment]
@rljacob - I think you're right about this:

> I don't think we have any tests with varying orbital params.

I missed the fact that #2561 only affects cesm.

> Do you want this in the release? I think it needs to be merged to your new maintenance branch.

Yes - this should come to master and the maintenance branch. @jedwards4b - I put the maint-5.6_bugfix tag on this; let me know if there's anything else that needs to be done for this to happen.

**Comment by billsacks** [comment]
Here are two tests that give answers that differ from baseline, as expected:

```
ERP_Ly3_P72x2.f10_f10_musgs.IHistClm50BgcCrop.cheyenne_intel.clm-cropMonthOutput
SMS_Ld5.f19_g17.IHistClm50Bgc.cheyenne_intel.clm-decStart
```

Note that both of these have orb_mode = variable_year and cross a year boundary (the decStart test starts Dec 30 and ends in early January).

**Comment by rljacob** [comment]
In that case the target for this PR should be the maintenance branch.

**Comment by billsacks** [comment]
Okay, thanks - just changed the base to the maintenance branch.

**Comment by rljacob** [comment]
Just merge the things you want in the release to the maint-5.6 branch and when you're done, merge the maint branch to master.

Never merge master to maint.

**Comment by rljacob** [comment]
Maybe the maint branch was made too early?  You don't really need that branch until there's a PR for master that you *don't* want in the release.

**Comment by jedwards4b** [comment]
@jgfouca wanted to bring in an e3sm merge, which is why the maint branch was made.

**Comment by rljacob** [comment]
But the PR for that isn't even up yet.

**Comment by jedwards4b** [comment]
True but I don't see any point in undoing what's done.

**Comment by rljacob** [comment]
Sure its just more hassle.   There has been at least one commit to master after the maint-5.6 banch point so its also important that feature branches intended for maint-5.6 start either at the branch point or from the maint-5.6 branch itself.

**Comment by rljacob** [comment]
Hang on:  it looks like the maint-5.6 branch does not start from master.  That's a problem.  Lets take this to Slack.

**Comment by billsacks** [comment]
@mvertens I can hit merge if you feel this is good to go... but can you first confirm that the statement I made about when answers will change is correct?

**Comment by mvertens** [comment]
@billsacks - your statement about when answers will change is correct.
